fvm/evm/handler/handler.go

```diff
@@ -219,6 +220,71 @@ func (h *ContractHandler) run(
 	return res, nil
 }
 
+func (h *ContractHandler) DryRun(
+	from types.Address,
+	to *types.Address,
+	gasLimit *types.GasLimit,
+	value types.Balance,
+	data []byte,
+) *types.ResultSummary {
+	res, err := h.dryRun(from, to, gasLimit, value, data)
+	panicOnError(err)
+	return res.ResultSummary()
+}
+
+func (h *ContractHandler) dryRun(
+	from types.Address,
+	to *types.Address,
+	gasLimit *types.GasLimit,
+	value types.Balance,
+	data []byte,
+) (*types.Result, error) {
+
+	ctx, err := h.getBlockContext()
+	if err != nil {
+		return nil, err
+	}
+
+	blk, err := h.emulator.NewBlockView(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	call := &types.DirectCall{
+		Type:    types.DirectCallTxType,
+		SubType: types.DryRunSubType,
+		From:    from,
+		Data:    data,
+		Value:   value,
+	}
+
+	if to != nil {
+		call.To = *to
+	}
+	if gasLimit != nil {
+		call.GasLimit = uint64(*gasLimit)
+	} else {
+		call.GasLimit = math.MaxUint
+	}
+
+	res, err := blk.DirectCall(call)
```

Comment:
I think a better way might be we expose a different function on block for dryRun and we don't use the direct call.

Reply:
I can do that yes, but then should I also create a new type to not use `types.DirectCall`? but that type would need basically all the same fields, because internally would be converted to a message anyway. Seems like a bit of duplication.

Reply:
I think the input can be any of the direct call types